pkg/common/common.go

```diff
@@ -313,31 +312,37 @@ func GetCustomQueries(keptnEvent *BaseKeptnEvent, logger *keptn.Logger) (map[str
 	return sliMap, nil
 }
 
-// GetDynatraceConfig loads dynatrace.conf for the current service
-func GetDynatraceConfig(keptnEvent *BaseKeptnEvent, logger *keptn.Logger) (*DynatraceConfigFile, error) {
+// GetDynatraceConfig loads dynatrace.conf for the current service.
+// If none is found, it returns a default configuration.
+func GetDynatraceConfig(keptnEvent *BaseKeptnEvent, logger *keptn.Logger) DynatraceConfigFile {
+	dynatraceConfFile := getBaseDynatraceConfig(keptnEvent, logger)
+	if dynatraceConfFile.DtCreds == "" {
+		dynatraceConfFile.DtCreds = "dynatrace"
+	}
+	// implementing https://github.com/keptn-contrib/dynatrace-sli-service/issues/90
+	dynatraceConfFile.DtCreds = ReplaceKeptnPlaceholders(dynatraceConfFile.DtCreds, keptnEvent)
+	return dynatraceConfFile
+}
 
-	dynatraceConfFileContent, err := GetKeptnResource(keptnEvent, DynatraceConfigFilename, logger)
+func getBaseDynatraceConfig(keptnEvent *BaseKeptnEvent, logger *keptn.Logger) DynatraceConfigFile {
 
-	if err != nil {
-		return nil, err
+	var defaultDynatraceConfigFile = DynatraceConfigFile{
```

Comment:
Your logic does not change the behavior: In case of any error (e.g. dynatrace.conf.yaml is invalid), we fall back to the default Dynatrace creds. 
But I am wondering if this is the expected behavior of a user. 
Can we please add a log message that the service is falling back to the default config?

Reply:
Currently if parsing the YAML fails an error will be reported, see comment below, line 335